lib/storage/batch.go

```diff
@@ -0,0 +1,130 @@
+package storage
+
+import (
+	"sync"
+
+	"github.com/syndtr/goleveldb/leveldb"
+	leveldbIterator "github.com/syndtr/goleveldb/leveldb/iterator"
+	leveldbOpt "github.com/syndtr/goleveldb/leveldb/opt"
+	leveldbUtil "github.com/syndtr/goleveldb/leveldb/util"
+)
+
+type BatchCore struct {
+	sync.RWMutex
+
+	core  LevelDBCore
+	batch *leveldb.Batch
+
+	inserted map[string][]byte
+}
+
+func NewBatchCore(core LevelDBCore) *BatchCore {
+	return &BatchCore{
+		core:     core,
+		batch:    &leveldb.Batch{},
+		inserted: map[string][]byte{},
+	}
+}
+
+func (bb *BatchCore) convertKey(key []byte) string {
+	return string(key)
+}
+
+func (bb *BatchCore) Has(key []byte, opt *leveldbOpt.ReadOptions) (bool, error) {
+	bb.RLock()
+	defer bb.RUnlock()
+
+	var found bool
+	if _, found = bb.inserted[bb.convertKey(key)]; found {
```

Comment:
I figure just using `string(key)` is better to understand :)

Reply:
@Charleslee522 I see. I will.